api/scan_test.go

```diff
@@ -0,0 +1,137 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tsuru/cst/scan"
+	"github.com/tsuru/cst/scan/schd"
+
+	"github.com/labstack/echo"
+)
+
+func TestCreateScan(t *testing.T) {
+
+	t.Run(`When payload is empty, should return a bad request response`, func(t *testing.T) {
+
+		e := echo.New()
+
+		request := httptest.NewRequest(http.MethodPost, "/", strings.NewReader(``))
+		request.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+		recorder := httptest.NewRecorder()
+
+		context := e.NewContext(request, recorder)
+
+		err := createScan(context)
+
+		if assert.Error(t, err) {
```

Comment:
No need to change this, more of a tip for the future. You can avoid the `if` statement by using `require.Error(t, err)` from `github.com/stretchr/testify/require`. It will fail the test and return immediately so the following lines will never be executed.

Reply:
wow! I'll use that in the next commits. Thanks for the advice.